src/media/ros/ros_reader.cpp

```diff
@@ -1434,35 +1439,34 @@ namespace librealsense
         return std::make_pair(id, std::make_shared<const_value_option>(description, value));
     }
 
-    std::shared_ptr<librealsense::processing_block_interface> ros_reader::create_processing_block(const rosbag::MessageInstance& value_message_instance, bool& depth_to_disparity, std::shared_ptr<options_interface> options)
+    std::shared_ptr< librealsense::processing_block_interface >
+    ros_reader::create_processing_block( const rosbag::MessageInstance & value_message_instance,
+                                         bool & depth_to_disparity,
+                                         std::shared_ptr< options_interface > sensor_options )
     {
         auto processing_block_msg = instantiate_msg<std_msgs::String>(value_message_instance);
-        rs2_extension id;
-        convert(processing_block_msg->data, id);
-        std::shared_ptr<librealsense::processing_block_interface> disparity;
-
-        switch (id)
-        {
-        case RS2_EXTENSION_DECIMATION_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_DECIMATION_FILTER>::type>();
-        case RS2_EXTENSION_THRESHOLD_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_THRESHOLD_FILTER>::type>();
-        case RS2_EXTENSION_DISPARITY_FILTER:
-            disparity = std::make_shared<ExtensionToType<RS2_EXTENSION_DISPARITY_FILTER>::type>(depth_to_disparity);
-            depth_to_disparity = false;
-            return disparity;
-        case RS2_EXTENSION_SPATIAL_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_SPATIAL_FILTER>::type>();
-        case RS2_EXTENSION_TEMPORAL_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_TEMPORAL_FILTER>::type>();
-        case RS2_EXTENSION_HOLE_FILLING_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_HOLE_FILLING_FILTER>::type>();
-        case RS2_EXTENSION_HDR_MERGE:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_HDR_MERGE>::type>();
-        case RS2_EXTENSION_SEQUENCE_ID_FILTER:
-            return std::make_shared<ExtensionToType<RS2_EXTENSION_SEQUENCE_ID_FILTER>::type>();
-        default:
-            return nullptr;
+        std::string name = processing_block_msg->data;
+        if( name == "Disparity Filter" )
```

Comment:
I must say I dont like when enums are replaced with json strings.
Yes it is more dynamic, but it makes it super hard to debug and add bugs.

if you are new to this code you will work hard to find an issue..
Please try to keep it DDS only

Reply:
Not sure I understand.
The old code used strings (extension names) to identify blocks. I am not changing the format. We still use strings.
But now, rather than the extension name, we use the block name, which is more specific.
I don't understand what you mean by "json strings" -- there's no json here.

Reply:
Line 1441 on the left:
`convert(processing_block_msg->data, id);`
Where `processing_block_msg->data` is a string.
`convert` is in `ros_file_format.h`:
```
    template <typename T>
    inline bool convert(const std::string& source, T& target)
    {
        if (!try_parse(source, target))
        {
            LOG_INFO("Failed to convert source: " << source << " to matching " << typeid(T).name());
            return false;
        }
        return true;
    }
```
And it calls `try_parse` which is a loop over all our known extension types.